osquery/tables/system/windows/registry.h

```diff
@@ -0,0 +1,160 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#pragma once
+
+#include <WbemIdl.h>
+#include <Windows.h>
+#include <intrin.h>
+#include <iostream>
+#include <osquery/tables.h>
+#include <string>
+#include <vector>
+
+namespace osquery {
+namespace tables {
+
+// Windows registry helper macros for key and path sizes
+#define MAX_KEY_LENGTH 255
+#define MAX_VALUE_NAME 16383
+
+/// Takes a Windows FILETIME object and returns seconds since epoch
+static LONGLONG filetimeToUnixtime(FILETIME ft) {
+  LARGE_INTEGER date, adjust;
+  date.HighPart = ft.dwHighDateTime;
+  date.LowPart = ft.dwLowDateTime;
+  adjust.QuadPart = 11644473600000 * 10000;
+  date.QuadPart -= adjust.QuadPart;
+  return date.QuadPart / 10000000;
+};
+
+/// Microsoft helper function for getting the contents of a registry key
+std::vector<std::map<std::string, std::string>> queryKey(HKEY hKey) {
+  TCHAR achClass[MAX_PATH] = TEXT("");
+  DWORD cchClassName = MAX_PATH;
+  DWORD cSubKeys = 0;
+  DWORD cbMaxSubKey;
+  DWORD cchMaxClass;
+  DWORD cValues;
+  DWORD cchMaxValueName;
+  DWORD cbMaxValueData;
+  DWORD cbSecurityDescriptor;
+  DWORD retCode;
+  FILETIME ftLastWriteTime;
+  retCode = RegQueryInfoKey(hKey,
+                            achClass,
+                            &cchClassName,
+                            nullptr,
+                            &cSubKeys,
+                            &cbMaxSubKey,
+                            &cchMaxClass,
+                            &cValues,
+                            &cchMaxValueName,
+                            &cbMaxValueData,
+                            &cbSecurityDescriptor,
+                            &ftLastWriteTime);
+
+  std::vector<std::map<std::string, std::string>> queryResults;
+  TCHAR achKey[MAX_KEY_LENGTH];
+  DWORD cbName;
+  if (cSubKeys > 0) {
+    for (auto i = 0; i < cSubKeys; i++) {
+      cbName = MAX_KEY_LENGTH;
+      retCode = RegEnumKeyEx(hKey, i, achKey, &cbName, nullptr, nullptr,
+                             nullptr, &ftLastWriteTime);
+      if (retCode != ERROR_SUCCESS) {
+        continue;
+      }
+      Row r;
+      r["reg_type"] = "subkey";
+      r["index"] = std::to_string(i);
+      r["subkey"] = achKey;
+      r["mtime"] = std::to_string(filetimeToUnixtime(ftLastWriteTime));
+      queryResults.push_back(r);
+    }
+  }
+
+  if (cValues <= 0) {
```

Comment:
Awesome, when you end the method early, you save all of that stack from `achValue`!